File: php/testcode/xss_029.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_js_template_literal
function xss029(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('val');
    $html = "<script>var x = `$input`</script>"; // vuln-code-snippet vuln-line php_xss_js_template_literal
    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_js_template_literal

File: rust/testcode/xss_029.rs
//! CWE-79: User input embedded inside an SVG text element without escaping.

// vuln-code-snippet start testcodeXss029
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let label = req.param("label");

    let html = format!("<svg><text x='10' y='20'>{}</text></svg>", label); // vuln-code-snippet target-line testcodeXss029

    super::shared::BenchmarkResponse::ok(&html)
}
// vuln-code-snippet end testcodeXss029
